File: exercise/kotlin/day15/src/main/kotlin/santaChristmasList/operations/Dependencies.kt
package santaChristmasList.operations

class Factory : HashMap<Gift, ManufacturedGift>() {
    fun findManufacturedGift(gift: Gift): ManufacturedGift? {
        return this[gift]
    }
}

class Inventory : HashMap<String, Gift>() {
    fun pickUpGift(barCode: String): Gift? {
        return this[barCode]
    }
}

class WishList : HashMap<Child, Gift>() {
    fun identifyGift(child: Child): Gift? {
        return this[child]
    }
}

File: exercise/scala/day15/src/main/scala/santaChristmasList/operations/Dependencies.scala
package santaChristmasList.operations

class Factory extends collection.mutable.HashMap[Gift, ManufacturedGift] {
  def findManufacturedGift(gift: Gift): Option[ManufacturedGift] = {
    this.get(gift)
  }
}

class Inventory extends collection.mutable.HashMap[String, Gift] {
  def pickUpGift(barCode: String): Option[Gift] = {
    this.get(barCode)
  }
}

class WishList extends collection.mutable.HashMap[Child, Gift] {
  def identifyGift(child: Child): Option[Gift] = {
    this.get(child)
  }
}